### ISSUE TITLE
Blueprint scripting with DReyeVR

### ISSUE THREAD

**Issue by StijnOosterlinck** [HARPLab/DReyeVR#150]
Hi,

I was wondering whether we can use blueprint scripting together with DReyeVR?

I was especially looking for the functionality of "input keys" (from keyboard/logitech wheel).

Currently, I'm able to display a pause menu after a certain time, and remove it again after a certain time.
However, ideally, the pause menu should only be removed when a button on the logitech wheel is pressed (e.g. the 'A' key).
But using key input events seems not to work (also not with normal keyboard keys), is this because DReyeVR has already set bindings for all these keys?

I would like to give some functionality to some keys using the blueprints, and for this I'm also supposed to enable a setting called 'auto receive input' for a certain actor. For which actor should I set this then?

The scripts I am running now are just made in the blueprint class of the TeslaM3, but enabling this option here has no effect.

Has anyone experience with this?
Thanks in advance!

**Comment by ajdroid** [comment]
Combining blueprint with C++ is doable but kinda messy and we don't recommend it. Instead you should try using the input handling logic we have in dreyevr pawn which does work and should be easily extendable so everything can be done in C++

**Comment by StijnOosterlinck** [comment]
Thanks for your answer @ajdroid ,

The problem is that I'm trying to combine the input handling with other things (such as showing a pause menu etc). Based on previous issues (like [this one](https://github.com/HARPLab/DReyeVR/issues/99)), I concluded that pausing the game through code seemed complicated (especially if you want to register inputs while the game is paused).

Because of this I tried it the blueprint-way and pausing the game using blueprints is possible so it seems more logical to do the input handling logic (only for a button click in the menu), also in this blueprint 'event-graph'. However, the issue I mentioned also discussed the fact that the whole simulation could be frozen, so no inputs can be registered, so if this is the case with the blueprint-method as well then this problem is not relevant anymore..

I can use C++ to handle everything that needs to happen when a button is clicked, but I just want to register it in the blueprint event graph as well so that the pause menu can be removed when the button is clicked, would this not be possible or is this already too messy?
